src/Components/HeaderMain.jsx

```diff
@@ -1,90 +1,161 @@
-import React from 'react';
-import HeaderCategory from './HeaderCategory'
-import { Menu, Header, Select, Container } from 'semantic-ui-react'
-import { Link } from 'react-router-dom'
-
-const countryOptions = [
-  {
-    key: "Sweden",
-    text: "Sweden",
-    value: "Sweden",
-  },
-]
-
-const cityOptions = [
-  {
-    key: "Stockholm",
-    text: "Stockholm",
-    value: "Stockholm",
-  },
-]
-
-const mainLabels = [
-  {
-    name: 'Write An Article',
-    link: '/write-article',
-    id: 'write_article'
-  }, {
-    name: 'Review Articles',
-    link: '/review-articles',
-    id: 'review_articles'
+import React, { Component } from 'react';
+import HeaderCategory from './HeaderCategory';
+import { Menu, Header, Select, Container, Divider, Segment } from 'semantic-ui-react';
+import { Link } from 'react-router-dom';
+import { connect } from 'react-redux';
+import { signOutUser } from '../reduxTokenAuthConfig';
+
+
+class HeaderMain extends Component {
+
+  signOut = (e) => {
+    e.preventDefault()
+    const { signOutUser } = this.props
+    signOutUser()
+      .then(response => {
+        window.location.reload(true);
```

Comment:
Why are we doing this?

Reply:
cause i tried to redirect using history.push like the other redux methods and i was getting an error.

Reply:
tried with {Link} and {Redirect} also with no luck

Reply:
Are you rendering this component in a Route?
If not then the props.history object is not available out of the box. You need to import another hoc called withRouter form react-router-dom, which gives you access to the history object.

I am just highlighting it here because it is generally a bad practice to cause reloads in a React application, which could trigger a lot of other flows unnecessarily plus you will lose the current state.

Reply:
WoW!
I didn't expect it to be so easy.
Another thing learned. Thanks a lot man!